### PR TITLE
feat: Export event as xCalendar

### LINKED ISSUE #4849: Add feature to export event as xCalendar
**Is your feature request related to a problem? Please describe.**
This feature is missing in version 2.

**Describe the solution you'd like**
Follow the steps of version 1.

**Describe alternatives you've considered**
\-

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #4849 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Add the feature to export event as xCalendar

#### Changes proposed in this pull request:
- Followed the steps of version 1

### CODE REVIEW

File: utils/common.py [hunk 1]
```diff
@@ -39,3 +39,14 @@ def wrapper(k, *args, **kwargs):
         setattr(k, "make_object", eval("make_object", *args, **kwargs))
         return k
     return wrapper
+
+
+def format_timedelta(td):
```
Comment (schedutron) [inline utils/common.py]: Can't we simply use `str(td)`? For example, I get a clean formatted output for the following:
```
>>> import datetime
>>> start = datetime.datetime(2018, 6, 20, 11, 0)
>>> end = datetime.datetime(2018, 6, 20, 12, 9)
>>> delta = end - start
>>> str(delta)
'1:09:00'
```

Comment (mayank8318) [reply utils/common.py]: Yes, that should work too. Let me check

File: app/api/exports.py [hunk 2]
```diff
@@ -65,6 +65,28 @@ def export_download(event_id, path):
     return response
 
 
+@export_routes.route('/events/<string:event_identifier>/export/xcal', methods=['GET'])
```
Comment (Kreijstal) [inline app/api/exports.py]: Is there any difference between event_identifier and event_id?

File: app/api/helpers/tasks.py [hunk 3]
```diff
@@ -132,3 +133,28 @@ def export_ical_task(self, event_id):
...
+
+@celery.task(base=RequestContextTask, name='export.xcal', bind=True)
+def export_xcal_task(self, event_id):
+    event = db.session.query(Event).filter_by(id=event_id).first()
+
+    try:
+        filedir = current_app.config.get('BASE_DIR') + '/static/uploads/temp/'
+        if not os.path.isdir(filedir):
+            os.makedirs(filedir)
+        filename = "xcal.xcs"
```
Comment (Kreijstal) [inline app/api/helpers/tasks.py]: if two people export different events at the same time does it get overwritten?

File: app/api/exports.py (hunk 2 above)
Comment (mayank8318) [reply app/api/exports.py]: Yes, in terms of the database id is the auto-generated integer key and event_identifier and identifier is the string hash i think

Comment (mayank8318) [reply app/api/exports.py]: Front-end uses the string hash(event_identifier) everywhere

**Pull request merged**